shell/platform/android/test/io/flutter/embedding/engine/systemchannels/PlatformChannelTest.java

```diff
@@ -26,18 +26,53 @@
 @RunWith(RobolectricTestRunner.class)
 public class PlatformChannelTest {
     @Test
-    public void setSystemExclusionRectsSendsSuccessMessageToFramework() throws JSONException {
+    public void itSendsSuccessMessageToFrameworkWhenSettingSystemGestureExclusionRects() throws JSONException {
+        // --- Test Setup ---
         DartExecutor dartExecutor = mock(DartExecutor.class);
         PlatformChannel platformChannel = new PlatformChannel(dartExecutor);
         PlatformMessageHandler platformMessageHandler = mock(PlatformMessageHandler.class);
         platformChannel.setPlatformMessageHandler(platformMessageHandler);
+        Result result = mock(Result.class);
+
+        int top = 0;
+        int right = 500;
+        int bottom = 250;
+        int left = 0;
+
+        JSONObject JsonRect = new JSONObject();
+        JsonRect.put("top", top);
+        JsonRect.put("right", right);
+        JsonRect.put("bottom", bottom);
+        JsonRect.put("left", left);
+        JSONArray inputRects = new JSONArray();
+        inputRects.put(JsonRect);
+
+        MethodCall callSystemGestureExclusionRects = new MethodCall(
+            "SystemGestures.setSystemGestureExclusionRects",
+            inputRects
+        );
+
+        // --- Execute Test ---
+        platformChannel.parsingMethodCallHandler.onMethodCall(callSystemGestureExclusionRects, result);
+
+        // --- Verify Results ---
+        verify(result, times(1)).success(null);
+    }
+
+    @Test
+    public void itProperlyDecodesGestureRectsWhenSettingSystemGestureExclusionRects() throws JSONException {
+        // --- Test Setup ---
+        DartExecutor dartExecutor = mock(DartExecutor.class);
+        PlatformChannel platformChannel = new PlatformChannel(dartExecutor);
+        PlatformMessageHandler platformMessageHandler = mock(PlatformMessageHandler.class);
+        platformChannel.setPlatformMessageHandler(platformMessageHandler);
+        Result result = mock(Result.class);
 
         int top = 0;
         int right = 500;
         int bottom = 250;
         int left = 0;
 
-        ResultsMock resultsMock = mock(ResultsMock.class);
         JSONObject JsonRect = new JSONObject();
         JsonRect.put("top", top);
         JsonRect.put("right", right);
```

Comment:
I can't comment on the right line, but let's consider an improved name for `inputRects` here as well.